src/server/hook_func.c

```diff
@@ -6471,22 +6471,25 @@ add_pending_mom_allhooks_action(void *minfo, unsigned int action)
 void
 next_sync_mom_hookfiles(void)
 {
-	int	timeout_sec;
+	long	timeout_sec;
 	time_t	timeout_time;
 	time_t	current_time;
+	int	timed_out = 0;
 
 	if (pbs_conf.pbs_use_tcp == 1) {
-		timeout_sec = SYNC_MOM_HOOKFILES_TIMEOUT_TPP;
+		if (server.sv_attr[(int)SRV_ATR_sync_mom_hookfiles_timeout].at_flags & ATR_VFLAG_SET) {
+			timeout_sec = server.sv_attr[(int)SRV_ATR_sync_mom_hookfiles_timeout].at_val.at_long;
+		} else {
+			timeout_sec = SYNC_MOM_HOOKFILES_TIMEOUT_TPP;
+		}
 	} else {
 		timeout_sec = SYNC_MOM_HOOKFILES_TIMEOUT;
 	}
 	current_time = time((time_t *) 0);
 	timeout_time = g_sync_hook_time + timeout_sec;
-	if (do_sync_mom_hookfiles && 
-		(!sync_mom_hookfiles_proc_running ||
-		 	(current_time > timeout_time))) {
 
-		if (sync_mom_hookfiles_proc_running) {
+	if (sync_mom_hookfiles_proc_running) {
+		if (current_time > timeout_time) {
```

Comment:
The nice thing about a void function is that you can just return. Please check for any conditions where the function should return first. Then the remaining code doesn't need as much indentation.